src/rt/model.c

```diff
@@ -111,6 +111,7 @@ typedef struct _rt_model {
    memblock_t        *memblocks;
    model_thread_t    *threads[MAX_THREADS];
    ptr_list_t         eventsigs;
+   rt_assert_cnts_t   assert_cnts;
```

Comment:
```suggestion
   unsigned   assert_cnts[SEVERITY_FAILURE + 1];
```

That way you don't need all the case statements.